### PR TITLE
docs: update progressive rollout docs

### PR DESCRIPTION
**Pull request by Light2Dark**
## Description

Update the example used and add a description of how the configuration works. (updated)
Screenshot 2024-10-06 at 6 46 12 PM

## Closes issue(s)

Resolve https://github.com/thomaspoignant/go-feature-flag/issues/2452

## Checklist
- [ ] I have tested this code
- [ ] I have added unit test to cover this code
- [X] I have updated the documentation (`README.md` and `/website/docs`)
- [ ] I have followed the [contributing guide](CONTRIBUTING.md)

### CODE REVIEW

**Comment by Light2Dark** [comment]
Hey @thomaspoignant , I faced this issue and wanted to contribute to the docs. I think it might be worth to explain a little more on changing the percentage field. However, it is unintuitive. The following is my proposal to be added to the **Example** section:

### Using the percentage field (advanced)
If you do not intend to completely rollout a feature, you can modify the percentage field.

```yaml
    progressiveRollout:
      initial:
        variation: variationA
        percentage: 20   
        date: 2024-01-01T05:00:00.100Z
      end:
        variation: variationB
        percentage: 80
        date: 2024-01-05T05:00:00.100Z
```

- Before the `initial` date, the flag will return **variationB 20% of the time** and **variationA 80% of the time**.
- Between the `initial` and `end` date, the flag will progressively change from returning variationA to variationB.
- After the `end` date, the flag will return **variationB 80% of the time** and **variationA 20% of the time**.

File: website/docs/configure_flag/rollout/progressive.mdx [hunk 1]
```diff
@@ -71,27 +68,30 @@ variationA = "A"
...
+|------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **`percentage`**<br/>*(optional)*  | It represents the ramp of progress, at which level the flag starts (`initial`) and ends (`end`).<br/>**Default: `initial` = `0` and `end` = `100`**.                                                                                                                                                                                                                                                                                                                                                                                              |
```
Comment (thomaspoignant) [inline website/docs/configure_flag/rollout/progressive.mdx:97]: We probably need to change this table with explaining what should be in initial and end.

**Review by thomaspoignant — COMMENTED** [review]
@Light2Dark thanks a lot for this improvement of the documentation this is really great.

I agree that your proposal for the example is better to understand how it works, just to be sure you want to have 2 different examples? One without percentage _`(which means that we go from 0 to 100)_ and one with it?
I like your proposal, the bullet list helps a lot to understand better how it works.

The only thing I would change is the `Configuration fields` table that is not reflecting the format, I would prefer to have an entry for `initial` and one for `end`.

**Comment by Light2Dark** [comment]
> I agree that your proposal for the example is better to understand how it works, just to be sure you want to have 2 different examples? One without percentage _`(which means that we go from 0 to 100)_ and one with it? I like your proposal, the bullet list helps a lot to understand better how it works.

Yeap, 2 examples. One is a simple use case, and another for more advanced use. Let me make some changes and we can review then :)

> The only thing I would change is the `Configuration fields` table that is not reflecting the format, I would prefer to have an entry for `initial` and one for `end`.

Noted, let me add to this.

**Review by thomaspoignant — APPROVED** [review]

**Pull request merged**